internal/testrunner/runners/system/runner.go

```diff
@@ -51,6 +54,13 @@ const (
 	waitForDataDefaultTimeout = 10 * time.Minute
 )
 
+var errorPatterns = map[string][]string{
+	"elastic-agent": []string{
+		"Cannot index event publisher.Event",
+		"(panic|runtime error)",
```

Comment:
Nit. Pre-compile regular expressions here.
```suggestion
		regexp.MustCompile("Cannot index event publisher.Event"),
		regexp.MustCompile("(panic|runtime error)"),
```